docs: Create performance-summary.md for NeMo RL (#1560)
NVIDIA-NeMo/RL: docs/about/performance-summary.md

# Performance

As part of the NVIDIA NeMo Framework, NeMo RL, provides optimal performance for reinforcement learning on generative AI models by incorporating the latest optimizations - such as refit optimizations, mixed-precision training, and off-policy training.

This page provides performance benchmarks for LLMs and VLMs using NeMo RL across different GPU systems and configurations.

## Nomenclature

- **GBS**: Global Batch Size
- **MBS**: Micro Batch Size
- **FSDP**: Fully Sharded Data Parallel
  - FSDP = 1: use FSDP

>   - FSDP = 0: use DDP (Distributed Data Parallel)

Reviewer: ```suggestion
```

Reviewer: this is not used in the following table

- **TP**: Tensor Parallel Size
- **PP**: Pipeline Parallel Size
- **CP**: Context Parallel Size
- **VP**: Virtual Pipeline Parallel Size
- **EP**: Expert Parallel Size
- **GA**: Number of Gradient Accumulations
- **T-**: Training related
- **G-**: Generation related
- **Training backend**: NeMo RL have two training backends: Megatron and PyTorch DTensor

## Performance Metrics

Since reinforcement learning consists of training, generation and transition between the two, performance measurement also reflects this. Specifically, we track the following metrics:
- **Step time**: Time for each step, which includes training, generation, policy logprobs, and refit time.

> - **Tokens/sec/GPU**: [insert formula here]

Reviewer: ```suggestion
- **Tokens/sec/GPU**: Total number of tokens used in training in a batch, divided by the step time, divided by the total number of GPUs.
```

- **Training MFU**: Model floating-point operations per second per GPU

## Performance Summary for Large Language Models

> Below are performance benchmarks for various large language models organized by release version. These results were obtained using performance recipes available [insert script link here].

Reviewer: ```suggestion
Below are performance benchmarks for various large language models organized by release version. These results were obtained using performance recipes available [here](https://github.com/NVIDIA-NeMo/RL/tree/r0.4.0/examples/configs/recipes/llm/performance).
```

- **Precision Options**: Performance comparisons between different precision modes (BF16, FP8)

---

## Nemo RL v0.4

### GRPO. Dataset: [OpenMathInstruct-2](https://huggingface.co/datasets/nvidia/OpenMathInstruct-2)

#### System: DGX-H100. Precision: Training BF16, Generation BF16. Training Backend: Megatron-core.

| Model | On/Off policy|T-Max Sequence Length | G-Average Seq len| #-GPUs | G-GBS |T-GBS|Generation [TP,PP]|Training [TP,CP,EP,PP,VPP] | Tokens / sec / GPU | Total Step time(s) |
|-------|--------|-----|-----|-----------------|------|----|----|----|-|-|
| LLAMA3.1_8B|On policy|4,096 | 1,060 | 16|2,048|512| [1,1] | [1,1,1,1,1,2,n/a]| 1,562| 97.7 |
| LLAMA3.1_8B|1-step Off|4,096 | 1,129 | 16|2,048|512| [1,1] | [1,1,1,1,1,2,n/a]| 2,161| 74.6 |
| DeepSeek V3 |On policy|1,536 |745|256|512|512| [32,1]|[1,1,16,16,n/a]| 11| 154|
| DeepSeek V3 |1-step Off|1,536 |744|512|512|512| [32,1]|[1,1,16,16,n/a]| 11.0| 77.9|
| Qwen3-235B |On policy|8,192|5,671|128|512|512| [16,1]|[2,2,16,8,n/a]| 45.7| 506|
| Qwen3-235B |1-step Off|8,192|5,691|256|512|512| [8,1]|[4,1,16,8,n/a]| 52.2| 241|
| Qwen3-30B3A |On policy|4,096|3,154|32|2,048|512| [2,1]|[1,1,8,1,n/a]| 925| 225|

> | Qwen3-30B3A |1-step Off|4,096|3,158|32|2,048|512| [2,1]|[1,1,8,2,n/a]| 864| 244|

Reviewer: The summary page config is different from the v0.4 test config
